### PR TITLE
update all components and refactor it , add real time search in pursh…

### PR DESCRIPTION
**Pull request by moayad-tarig**
update all components and refactor it , add real time search in purseasement component

### CODE REVIEW

File: app/Http/Livewire/SearchVendor.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public function decrementHighlight()
+    {
+        if ($this->highlightIndex === 0) {
+            $this->highlightIndex = count($this->vendors) - 1;
+            return;
+        }
+        $this->highlightIndex--;
+    }
+ 
+    public function selectContact()
```
Comment (owiesnama) [inline app/Http/Livewire/SearchVendor.php]: selectRecord would be a valid name instead on contacts

File: app/Http/Livewire/SearchVendor.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Http\Livewire;
+
+use App\Models\Vendor;
+use Livewire\Component;
+
+class SearchVendor extends Component
```
Comment (owiesnama) [inline app/Http/Livewire/SearchVendor.php]: Component name should be a generic name like searchDropdown

File: app/Http/Livewire/WarehouseItems.php [hunk 3]
```diff
@@ -15,7 +15,8 @@ class WarehouseItems extends Component
     public $isAddingNewItem = false;
     public $isUpdating = false;
     public $isDeleting = false;
-    public ItemsWarehouse $stock;
+    // public $ItemsWarehouse = $stock;
+    public $ItemsWarehouse;
```
Comment (owiesnama) [inline app/Http/Livewire/WarehouseItems.php:21]: I think this is typo, if not please change to lowercase

File: app/Http/Livewire/WarehouseItems.php [hunk 4]
```diff
@@ -15,7 +15,8 @@ class WarehouseItems extends Component
...
     public $stockToDelete;
     public $perPage = 10;
```
Comment (owiesnama) [inline app/Http/Livewire/WarehouseItems.php]: switch to WithSorting trait here

**Review by owiesnama — CHANGES_REQUESTED** [review]

File: app/Http/Livewire/WarehouseItems.php (hunk 3 above)
Comment (moayad-tarig) [reply app/Http/Livewire/WarehouseItems.php:21]: this is your code , not mine .

File: app/Http/Livewire/WarehouseItems.php (hunk 4 above)
Comment (moayad-tarig) [reply app/Http/Livewire/WarehouseItems.php]: ok

**Pull request merged**